### PR TITLE
Navbar

### LINKED ISSUE #4: Navigation bar
Add navigation bar that displays a title and a sign out button when logged in.

### PR DESCRIPTION
**Pull request by Aaron-Goldman**
Added navigation bar that displays a site title and a Sign Out button when logged in.

Should be merged after #3.

close #4

### CODE REVIEW

File: src/NavBar.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  return (
+    <AppBar position="static">
+      <Toolbar>
+        <Typography variant="h6" color="inherit" style={{ flexGrow: 1 }}>
+          Job Portal
+        </Typography>
+        <div>
+          {signOutButton}
+        </div>
```
Comment (ahmed-gz) [inline src/NavBar.js]: Unless I'm missing something, you don't need extra divs here, you can simply render the button only when logged in:
```
const isLoggedIn = !!auth.user;
//...
<Typography variant="h6" color="inherit" style={{ flexGrow: 1 }}>
  Job Portal
</Typography>
{isLoggedIn && <Button onClick={handleSignOut}>Sign Out</Button>}
```

**Review by ahmed-gz — CHANGES_REQUESTED** [review]
Looks ok overall, I would recommend you always open the PR against the branch it's depending on, for example if this PR should only get merged once  [#3](https://github.com/Aaron-Goldman/internship-job-portal/pull/3) is merged, then you should create your branch based of `auth` and open the PR against it, then once the other PR is merged you can simply change the target to master. Otherwise the PR will show changes from the other PR and it will be hard to review.

**Comment by Aaron-Goldman** [comment]
> Looks ok overall, I would recommend you always open the PR against the branch it's depending on, for example if this PR should only get merged once [#3](https://github.com/Aaron-Goldman/internship-job-portal/pull/3) is merged, then you should create your branch based of `auth` and open the PR against it, then once the other PR is merged you can simply change the target to master. Otherwise the PR will show changes from the other PR and it will be hard to review.

Should I do a `git rebase` now or will that just make things worse since I already opened this PR?

**Comment by ahmed-gz** [comment]
@Aaron-Goldman I think you will get merge conflicts no matter what you do once the other PR is merged to master, because you included change here that will be merged to master soon, so I would recommend either rebasing this PR and all others to `auth`  or simply start over if it's easier for you by creating new branches based on the previous branch that it's depending on, for example:

- Create `navbar` based on `auth`.
- Create `job-feed` based on `navbar`.
- Create `job-details` based on `job-feed`.

That way you would just need to sync each branch with it's parent once it's merged to master then delete the parent when you're done, with that I don't think you will ever get merge conflicts and each PR will only have the changes you made for that feature so you don't have to copy/paste stuff, does that make sense?

**Comment by ahmed-gz** [comment]
@Aaron-Goldman Ideally we want to achieve a clean diff between child and parent branches like this:

- https://github.com/Aaron-Goldman/internship-job-portal/compare/auth...Aaron-Goldman:navbar
- https://github.com/Aaron-Goldman/internship-job-portal/compare/navbar...Aaron-Goldman:job-feed
- https://github.com/Aaron-Goldman/internship-job-portal/compare/job-feed...Aaron-Goldman:job-details

Also please delete the branches that are no longer needed in your repo!

**Comment by Aaron-Goldman** [comment]
The PR branches should be fixed now.

File: src/NavBar.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  Toolbar,
+  Typography,
+} from '@material-ui/core';
+import { useAuth } from './AuthProvider';
+
+function NavBar() {
+  const auth = useAuth();
+  const handleSignOut = () => {
+    auth.signOut();
+  };
```
Comment (ahmed-gz) [inline src/NavBar.js]: You can make this shorter like this if you want: `const handleSignOut = () => auth.signOut();`

File: src/NavBar.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```
Comment (ahmed-gz) [inline src/NavBar.js]: Similar to other features, please move this file to `src/NavBar/index.js`

File: src/NavBar.test.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
```
Comment (ahmed-gz) [inline src/NavBar.test.js]: Similar to other features, please move this file to `src/NavBar/index.test.js`

**Review by ahmed-gz — CHANGES_REQUESTED** [review]
Looks good, left some suggestions!

File: src/App.js [hunk 5]
```diff
@@ -7,6 +7,7 @@ import {
   Route,
 } from 'react-router-dom';
 import apolloClient from './apollo';
+import NavBar from './Navbar';
```
Comment (ahmed-gz) [inline src/App.js]: For consistency let's use the same name for the file and its default export, preferably:
`import Navbar from './Navbar';`

File: src/Navbar/index.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import {
+  AppBar,
+  Button,
+  Toolbar,
+  Typography,
+} from '@material-ui/core';
+import { useAuth } from '../AuthProvider';
+
+function NavBar() {
```
Comment (ahmed-gz) [inline src/Navbar/index.js]: For consistency let's use the same name as the file `Navbar`

**Review by ahmed-gz — CHANGES_REQUESTED** [review]
Looks good, left one last suggestion!

**Review by ahmed-gz — APPROVED** [review]
Looks great, feel free to merge it 👍

**Pull request merged**